File: MusicManagerAPI/src/main/java/music/manager/model/UserMusicStatus.java
package music.manager.model;

import music.manager.constants.MusicTypes;
import org.springframework.context.annotation.Configuration;

import javax.persistence.*;
import java.util.List;
import java.util.Set;

/**
 * Created by ldalzotto on 25/04/2017.
 */
@Entity
@Table(name = "user_music_status", uniqueConstraints = @UniqueConstraint(columnNames = {"username"}))
public class UserMusicStatus {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    @Column(name = "id")
    private Long id;

    @Column(name = "username")
    private String username;

    @OneToMany(cascade = CascadeType.ALL, mappedBy = "userMusicStatus")
    private Set<MusicType> musicTypes;

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getUsername() {
        return username;
    }

    public void setUsername(String username) {
        this.username = username;
    }

    public Set<MusicType> getMusicTypes() {
        return musicTypes;
    }

    public void setMusicTypes(Set<MusicType> musicTypes) {
        this.musicTypes = musicTypes;
    }
}

File: MusicManagerAPI/src/scala/com/ldz/music/manager/model/UserMusicStatus.scala
package com.ldz.music.manager.model

import javax.persistence._

/**
  * Created by Loic on 20/06/2017.
  */
@Entity
@Table(name = "user_music_status", uniqueConstraints = Array(new UniqueConstraint(columnNames = Array("username"))))
class UserMusicStatus {


  @Id
  @GeneratedValue(strategy = GenerationType.AUTO)
  @Column(name = "id") private var id: Long = 0L
  @Column(name = "username") private var username: String = _
  @OneToMany(cascade = Array(CascadeType.ALL), mappedBy = "userMusicStatus")
  private var musicTypes: java.util.List[MusicType] = _

  def getId: Long = id

  def setId(id: Long): Unit = {
    this.id = id
  }

  def getUsername: String = username

  def setUsername(username: String): Unit = {
    this.username = username
  }

  def getMusicTypes: java.util.List[MusicType] = musicTypes

  def setMusicTypes(musicTypes: java.util.List[MusicType]): Unit = {
    this.musicTypes = musicTypes
  }
}
